types/pouchdb-mapreduce/index.d.ts

```diff
@@ -10,15 +10,19 @@
 /* tslint:disable:no-single-declare-module */
 
 declare namespace PouchDB {
+    type Map<Content extends {}> = (doc: Content) => void;
```

Comment:
Depends on perspective: slower because it will never run remotely on a CouchDB server, but faster than equivalent local map operation because function application is faster than an `eval` closure. (In that way the API documentation is partially misnamed/reversed: `eval` forces the creation of a fresh closure at runtime locally, but the function application does not; however, it has access to the ambient closure state of where it was written, which an `eval` version would not, so the name is correct from one point of view, but inaccurate from the other.)

Rambling aside: let's do it the easy way...